src/audio/module_adapter/module/iadk_modules.c

```diff
@@ -80,8 +83,33 @@ static int iadk_modules_init(struct processing_module *mod)
 	uint32_t instance_id = IPC4_INST_ID(mod->dev->ipc_config.id);
 	uint32_t log_handle = (uint32_t) mod->dev->drv->tctx;
 	/* Connect loadable module interfaces with module adapter entity. */
-	void *mod_adp = system_agent_start(md->module_entry_point, module_id,
-					   instance_id, 0, log_handle, &mod_cfg);
+	/* Check if native Zephyr lib is loaded */
+	struct sof_man_fw_desc *desc;
+
+	desc = lib_manager_get_library_module_desc(module_id);
+	if (!desc) {
+		comp_err(dev, "iadk_modules_init(): Failed to load manifest");
+		return -ENOMEM;
+	}
+	struct sof_man_module *module_entry =
+		    (struct sof_man_module *)((char *)desc + SOF_MAN_MODULE_OFFSET(0));
+
+	sof_module_build_info *mod_buildinfo =
+		(sof_module_build_info *)(module_entry->segment[SOF_MAN_SEGMENT_TEXT].v_base_addr);
+
+	void *mod_adp;
+
+	/* Check if module is FDK*/
+	if (mod_buildinfo->api_version_number.fields.major < _MAJOR_API_MODULE_VERSION) {
+		mod_adp = system_agent_start(md->module_entry_point, module_id,
+					     instance_id, 0, log_handle, &mod_cfg);
+	} else {
+		/* If not start agent for sof loadable */
+		mod->is_loadable = true;
```

Comment:
This is misleading isnt it? Whether a module is built with IADK or is a native SOF module, it is loadable nevertheless right? Do you want to call this mod->is_native_sof?